modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,151 +544,210 @@ sub print_diagram {
-                    print "Tweak.Changing\tpipeline.param[$param_name] ::\t$hash_pair->{'param_value'} --> $new_value_str\n";
+                    $tweakStructure->{Return}->{OldValue} = $hash_pair->{'param_value'};
+                    push @response, "Tweak.Changing  \tpipeline.param[$param_name] ::\t$hash_pair->{'param_value'} --> $new_value_str\n";
 
                     $hash_pair->{'param_value'} = $new_value_str;
                 } else {
-                    print "Tweak.Adding  \tpipeline.param[$param_name] ::\t(missing value) --> $new_value_str\n";
-
+                    $tweakStructure->{Return}->{OldValue} = undef;
+                    push @response, "Tweak.Adding  \tpipeline.param[$param_name] ::\t(missing value) --> $new_value_str\n";
                     $self->add_new_or_update( 'PipelineWideParameters',
                         'param_name'    => $param_name,
                         'param_value'   => $new_value_str,
                     );
                 }
             }
-
+          push @{$responseStructure->{Tweaks}}, $tweakStructure;
         } elsif($tweak=~/^pipeline\.(\w+)(\?|=(.+))$/) {
+            my $tweakStructure;
             my ($attrib_name, $operator, $new_value_str) = ($1, $2, $3);
+            $tweakStructure->{Object}->{Type} = 'Pipeline';
+            $tweakStructure->{Object}->{Id} = undef;
+            $tweakStructure->{Object}->{Name} = undef;
+            $tweakStructure->{Return}->{Field} = $attrib_name;
 
             if($self->can($attrib_name)) {
                 my $old_value = stringify( $self->$attrib_name() );
 
                 if($operator eq '?') {
-                    print "Tweak.Show    \tpipeline.$attrib_name ::\t$old_value\n";
+                    $tweakStructure->{Action} = 'SHOW';
+                    $tweakStructure->{Return}->{OldValue} = $old_value;
+                    $tweakStructure->{Return}->{NewValue} = $old_value;
+                    push @response, "Tweak.Show  \tpipeline.$attrib_name ::\t$old_value\n";
                 } else {
-                    print "Tweak.Changing\tpipeline.$attrib_name ::\t$old_value --> $new_value_str\n";
+                    $tweakStructure->{Action} = 'SET';
+                    $tweakStructure->{Return}->{OldValue} = $old_value;
+                    $tweakStructure->{Return}->{NewValue} = $new_value_str;
+                    push @response, "Tweak.Changing  \tpipeline.$attrib_name ::\t$old_value --> $new_value_str\n";
 
                     $self->$attrib_name( $new_value_str );
                     $need_write = 1;
                 }
 
             } else {
-                print "Tweak.Error   \tCould not find the pipeline-wide '$attrib_name' method\n";
+                if($operator eq '?') {
+                    $tweakStructure->{Action} = 'SHOW';
+                } else {
+                    $tweakStructure->{Action} = 'SET';
+                }
+                $tweakStructure->{Error} = 'Field not recognized';
```

Comment:
Since the same messages are going to be used for several errors, define the four messages outside of the function and use the variables here. It will be easier to maintain and will guarantee consistency.